### ISSUE TITLE
[PHP] Unable to use refs with the same name in different char cases within one alternative

### ISSUE THREAD

**Issue by KvanTTT** [antlr/antlr4#3462]
Consider the following grammar:

```g4
r: R r | X;
R: 'R';
X: 'X';
```

ANTLR correctly generates parser but PHP runtime reports

```
Fatal error: Cannot redeclare Context\RContext::r()
```

But it works correctly in other runtimes.

@marcospassos

**Comment by marcospassos** [comment]
That's probably because the token and rule names share the same name, and functions in PHP are case insensitive.

**Comment by KvanTTT** [comment]
It's true. But maybe it makes sense to change such behavior for consistency with other runtimes?

**Comment by marcospassos** [comment]
Maybe, but it's a breaking change. Also, it'll be hard to come with a prefix that matches all code styles. For example, if we decide to add a `token` prefix and developer name tokens like `TOKEN_NAME` it will result in methods like `tokenTOKEN_NAME`, which is terrible.

**Comment by KvanTTT** [comment]
Yes, it's a breaking change. Ok, it looks like it's better to leave it as is.

**Comment by kaby76** [comment]
Note: To test whether there are common lexer and parser symbol rules with the same name case insensitive, this works: `trparse *.g4 | trxgrep ' //(parserRuleSpec/RULE_REF/text() | lexerRuleSpec/TOKEN_REF/text())' | tr '[:upper:]' '[:lower:]' | sort | uniq -d`

**Comment by kaby76** [comment]
I ran this script on all grammars in the grammars-v4 repo and found a lot of these symbol conflicts.
```
directories=`find . -name pom.xml`
for i in $directories
do
  ls `dirname $i`/*.g4
  if [ "$?" != "0" ]
  then
    continue
  fi
  trparse `dirname $i`/*.g4 | trxgrep ' //(parserRuleSpec/RULE_REF/text() | lexerRuleSpec/TOKEN_REF/text())' | tr '[:upper:]' '[:lower:]' | sort | uniq -d
done
```

This is the output. 
[comm.txt](https://github.com/antlr/antlr4/files/10287865/comm.txt)

Looks like 972 renames. It might be good if the tool does the rename internally.
